libs/wingsdk/src/ex/dynamodb-table.ts

```diff
@@ -0,0 +1,396 @@
+   */
+  readonly primaryKey: string;
+}
+
+/**
+ * A cloud Dynamodb table.
+ *
+ * @inflight `@winglang/sdk.ex.IDynamodbTableClient`
+ */
+export abstract class DynamodbTable extends Resource {
+  /**
+   * Create a new DynamodbTable.
+   * @internal
+   */
+  public static _newDynamodbTable(
+    scope: Construct,
+    id: string,
+    props: DynamodbTableProps
+  ): DynamodbTable {
+    return App.of(scope).newAbstract(DYNAMODB_TABLE_FQN, scope, id, props);
+  }
+
+  /**
+   * Table name
+   */
+  public readonly name: string;
+  /**
+   * Table primary key name
+   */
+  public readonly primaryKey: string;
+
+  constructor(scope: Construct, id: string, props: DynamodbTableProps) {
+    super(scope, id);
+
+    Node.of(this).title = "DynamodbTable";
+    Node.of(this).description = "A DynamoDb Table";
+
+    if (!props.name) {
+      throw new Error("Dynamodb table name is not defined");
+    }
+    this.name = props.name;
+
+    if (!props.primaryKey) {
+      throw new Error("Dynamodb primary key is not defined");
+    }
+    this.primaryKey = props.primaryKey;
+  }
+
+  /** @internal */
+  public _getInflightOps(): string[] {
+    return [
+      DynamodbTableInflightMethods.PUT_ITEM,
+      DynamodbTableInflightMethods.UPDATE_ITEM,
+      DynamodbTableInflightMethods.DELETE_ITEM,
+      DynamodbTableInflightMethods.GET_ITEM,
+      DynamodbTableInflightMethods.SCAN,
+      DynamodbTableInflightMethods.TRANSACT_WRITE_ITEMS,
+    ];
+  }
+}
+
+/**
+ * Properties for `DynamodbTable.putItem`.
+ */
+export interface DynamodbTablePutItemProps {
+  /**
+   * A condition that must be satisfied in order for an operation to succeed.
+   * @default undefined
+   */
+  readonly conditionExpression?: string;
+}
+
+/**
+ * Properties for `DynamodbTable.updateItem`.
+ */
+export interface DynamodbTableUpdateItemProps {
+  /**
+   * An expression that defines one or more attributes to be updated.
+   * @default undefined
+   */
+  readonly updateExpression?: string;
+
+  /**
+   * One or more values that can be substituted in an expression.
+   * @default undefined
+   */
+  readonly expressionAttributeValues?: Json;
+}
+
+/**
+ * Properties for transact write item's update operation.
+ */
+export interface DynamodbTransactWriteItemPutProps {
+  /**
+   * The item to put.
+   */
+  readonly item: Json;
+  /**
+   * A condition that must be satisfied in order for an operation to succeed.
+   * @default undefined
+   */
+  readonly conditionExpression?: string;
+}
+
+/**
+ * Properties for transact write item's update operation.
+ */
+export interface DynamodbTransactWriteItemUpdateProps {
+  /**
+   * The item to update.
+   */
+  readonly key: string;
+  /**
+   * An expression that defines one or more attributes to be updated.
+   * @default undefined
+   */
+  readonly updateExpression?: string;
+  /**
+   * One or more values that can be substituted in an expression.
+   * @default undefined
+   */
+  readonly expressionAttributeValues?: Json;
```

Comment:
It can be `{ S: "value" }`, `{ N: "123" }`, `{ B: true }` and not sure if there's more. But if Wing needs to support the exported type, I guess we just can't use something else than Json.